sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonFileFormat.scala

```diff
@@ -114,7 +114,16 @@ class JsonFileFormat extends TextBasedFileFormat with DataSourceRegister {
     }
 
     (file: PartitionedFile) => {
-      val parser = new JacksonParser(actualSchema, parsedOptions)
+      // SPARK-21610: when the `requiredSchema` only contains `_corrupt_record`,
```

Comment:
I am actually rather -0 on this change. Both the current way and the previous way sound not quite compelling to me but the current way at least does arguably unnecessary parsing tries and we started to have this behaviour long time ago.. (at least I tried this in 1.6.3 and 1.5.2):

```scala
import org.apache.spark.sql.types._

val schema = new StructType().add("field", ByteType).add("_corrupt_record", StringType)
val file = "/tmp/sample.json"
val dfFromFile = sqlContext.read.schema(schema).json(file)
dfFromFile.filter($"_corrupt_record".isNotNull).count()
dfFromFile.filter($"_corrupt_record".isNull).count()
```